### PR TITLE
DB schema for creating table primarysite and cancertype.

### PR DESCRIPTION
**Pull request by Buwujiu**

### CODE REVIEW

File: src/main/resources/flyway/sql/V2__new_table_primarysite_cancertype.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  site_id               UUID  NOT NULL,
+  PRIMARY KEY(program_id, site_id),
+  FOREIGN KEY(program_id) REFERENCES PROGRAM(id),
+  FOREIGN KEY(site_id)  REFERENCES SITE(id)
+);
+
+ALTER TABLE PROGRAM ADD COLUMN date_updated TIMESTAMP NOT NULL;
+ALTER TABLE PROGRAM ADD COLUMN institutions VARCHAR(255) DEFAULT '';
+ALTER TABLE PROGRAM ADD COLUMN countries VARCHAR(255) NOT NULL;
+ALTER TABLE PROGRAM ADD COLUMN regions VARCHAR(255) DEFAULT '';
```
Comment (rtisma) [inline src/main/resources/flyway/sql/V2__new_table_primarysite_cancertype.sql]: according to the spec: spec

Regions has a zero-or-many to many programs relationship, meaning its optional
countries has a many to many relationship to programs, and there must be at least one country per program

this means the region and country tables must also be defined, along with their join tables

also, after a discussion with @rosibaj  we have decided to make `submittedDonors` and `genomicDonors` required and mutable, meaning they must be non null, and initialised to 0 upon creation. `Institutions` is a mutable and required field  (i.e non-null) that is a string. Refer to the [spec](https://wiki.oicr.on.ca/display/icgcargotech/Program+Service+-+Backend#ProgramService-Backend-3.1.1.1Program) for more details

after a discussion with @andricDu , we have agreed on a sql naming convention:
- table names should be lowercase, and snake case
- field names should be all lowercase and snake case
- all sql keywords (like `CREATE`, `NOT NULL`) should be uppercase. This will make reading the schema alot easier.

So in summary please do the following changes:
1. add the right constraints for submitted_donors, genomic_donors and institutions
2. create the region and country table and link them to the program
3. update all tables and field names to follow the sql naming convention

Comment (Buwujiu) [reply src/main/resources/flyway/sql/V2__new_table_primarysite_cancertype.sql]: As discussed with @rosibaj , we decided to keep Countries and Region as text fields in Program table. Will update ERD and specs later.

File: src/main/resources/flyway/sql/V2__new_table_primarysite_cancertype.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
+CREATE TYPE membership AS ENUM ('FULL', 'ASSOCIATE');
+
+CREATE TABLE cancer (
+    id                UUID PRIMARY KEY,
+    name              VARCHAR(255) UNIQUE NOT NULL
+);
+
+CREATE TABLE site  (
```
Comment (rtisma) [inline src/main/resources/flyway/sql/V2__new_table_primarysite_cancertype.sql]: Please change this to `primary_site`, as its more clear. Likewise, update the wiki
https://wiki.oicr.on.ca/pages/viewpage.action?spaceKey=icgcargotech&title=Program+Service+-+Backend

File: src/main/resources/flyway/sql/V2__new_table_primarysite_cancertype.sql [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+CREATE TABLE program_cancer (
+  program_id          UUID NOT NULL,
+  cancer_id           UUID NOT NULL,
+
+  PRIMARY KEY(program_id, cancer_id),
+  FOREIGN KEY(program_id) REFERENCES program(id),
+  FOREIGN KEY(cancer_id)  REFERENCES cancer(id)
+);
+
+CREATE TABLE program_site (
```
Comment (rtisma) [inline src/main/resources/flyway/sql/V2__new_table_primarysite_cancertype.sql]: this should be `program_primary_site`
